### ISSUE TITLE
Document UnitTest procedure

### ISSUE THREAD

**Issue by LFSaw** [supercollider/supercollider#3075]
PR #3017 will introduce that UnitTests are be explicitly mentioned in github PR templates. IMHO this requires developer guidance on how to 

+ write UnitTests --- partially described in the Documentation of the UnitTest Quark, though no mention of their location
+ install them --- currently, only outdated information in the UnitTest Quark, see #3016
+ run them --- should be part of/ linked at http://supercollider.github.io/contributing/ and the PR template prepared in #3017.

**Comment by LFSaw** [comment]
I am happy to help putting it together but I am currently not familiar with the procedure...

**Comment by LFSaw** [comment]
What needs to be done for a proper `UnitTest` guide :

+ [ ] remove outdated [CommonTests](https://github.com/supercollider-quarks/CommonTests) Quark from [master directory](https://github.com/supercollider-quarks/quarks/blob/master/directory.txt) --- how does one mark a quark as incompatible with an SC version e.g. > 3.7 ?
+ [ ] make UnitTest install/uninstall procedure as simple as possible from within sclang
```
"UnitTest".include;
"testsuite".include; // this could be e.g. a local-only quark that points to `$SC_PATH/testsuite/classlibrary`
// recompile
UnitTest.runAll; // or
UnitTest.gui; // or
UnitTest.findTestClass(TESTCLASS).run;
```
+ [ ] write a guideline on where to implement UnitTests for new functionality/classes

**Comment by brianlheim** [comment]
good point @LFSaw ! I don't think any of us are very familiar with the procedure TBH. I would really appreciate it if you could put some time in here! Testing is very important and we don't do enough of it.

**Comment by LFSaw** [comment]
does my above-outlined procedure make sense to you then? 
@telephon, since you seem to have written a lot of test cases recently, does this look familiar to you?

**Comment by brianlheim** [comment]
the to-do list you have is good IMO. i would be happy to comment on any guidelines for writing tests. there is probably a simpler way to include the test suite than by creating a separate quark for it.

**Comment by LFSaw** [comment]
I'll see if I can get a proposal up in the next days. But it means I need to dive into the `UnitTest` stuff... which is mostly (miss/un/outdated-)documented-land... so no guarantees. Whoever wants to jump on it, please do!

**Comment by LFSaw** [comment]
quick update... although it looks like I haven't done anything on this, well... I at least started looking into `UnitTest`. So, accumulating knowledge to be able to soon write documentation. Stay tuned.

**Comment by muellmusik** [comment]
This is WIP. @brianlheim let us know if/when you're happy with closing this.

**Comment by dyfer** [comment]
> I think this should replace the Writing Tests guide in schelp files. I'd volunteer to convert it to scdoc if others feel the same

I agree that "Writing Tests" in the help should be updated. It provides some basic information, but also has numerous typos and some inaccurate code (e.g. `"SCLANG_CONF".getenv` does not return the location of the config file, at least on macOS). But I would also want to avoid having that documentation live in two places (since it will inadvertently get out of sync eventually).

The wiki page says it's a work-in-progress document. Whether that still the case or not, I feel like having it being kept up to date on the wiki might be more efficient than having it in the help file.

I think I'd prefer if we:
- clean up "Writing Tests", correcting typos and removing inaccurate information
- provide a link to the wiki page close to the top of "Writing Tests"

Alternatively, we could move the wiki page to the help and then link to that from the wiki page (removing the wiki content to avoid having to update this information in two places down the line).

What do you think?
